tservice/src/seal.rs

```diff
@@ -70,17 +71,31 @@ impl<T: AsRef<[u8]> + Default> SealedBuilder<T> {
     ///
     /// # Arguments
     /// * `data` - The data to be encrypted/sealed
-    pub fn new(data: T) -> Self {
-        Self {
+    ///
+    /// # Errors
+    /// Returns `Error::InvalidParameter` ff `data` is empty or is not fully
+    /// within the enclaves memory space
+    pub fn new(data: T) -> Result<Self> {
```

Comment:
Since we are now doing the checking, it felt like doing it as soon as possible would be the most preferred to clients instead of waiting until the `build()` call.